src/Mod/PartDesign/fcgear/involute.py

```diff
@@ -58,23 +59,64 @@ def CreateExternalGear(w, m, Z, phi,
     Ra = Rpitch + addendum    # tip (addendum) circle radius
     Rroot = Rpitch - dedendum # root circle radius
     fRad = filletCoeff * m  # fillet radius, max 1.5*clearance
-    Rf = sqrt((Rroot + fRad)**2 - fRad**2) # radius at top of fillet
-    if (Rb < Rf):
-        Rf = Rroot + fRad/1.5 # fRad/1.5=clerance, with crearance=0.25*m
+    Rc = Rroot + fRad # radius at the center of the fillet circle
+    Rf = Rc  # radius at top of fillet, assuming fillet below involute
+    filletWithinInvolute = Rf > Rb # above the base circle we have the involute,
+                                   # below we have a straight line towards the center.
+    if (filletWithinInvolute):
+        # In this case we need tangency of the involute and the fillet circle.
+        # It has to be somewhere between max(Rb,Rroot) and Rc.
+        # So we need the radius r from the origin depending on the tangent angle, for both:
+        # - the involute: ri(t) = Rb * sqrt(1 + t**2), rationale: see below.
+        # - the fillet circle: a bit more complex, see below.
+        # and then find the r where both tangents are equal.
+        # As the tangent angle of the involute equals the parameter t, we can just take the
+        # parametric polar representation of the involute as our first equation.
+        # For a circle in parameter form, the tangent angle is pi/2 - t (for the first quadrant,
+        # i.e. 0 <= t <= pi/2, and more than one quadrant is not of interest for us). Unfortunately,
+        # the fillet circle is offset from the origin by Rc/phi, so things becomes more messy:
+        # rc(t) = sqrt((Rc*cos(phi) + fRad*cos(t))**2 + (Rc*sin(phi) + fRad*sin(t))**2)
+        # To get rid or any sin^2 and cos^2 we assume phi "very small", i.e. sin(phi) becomes 0.
+        # This is justyfied because Rc is much larger than fRad and the parallax error is
+        # neglectable. Next we substitute the parameter t by pi/2-q-pi. For one to account for the
+        # tangent angle definitions (see above), and then to turn our cicle by 180° as for the
+        # root fillet we need the third quadrant of the circle (NB: we are looking at the upper
+        # half of the right most tooth, i.e. the involute gorws downwards!). This results in
```

Comment:
gorws->grows

Reply:
Thanks again! I read over this so many times that by now I got completely blind ;-)